server/bleep/src/webserver/answer.rs

```diff
@@ -596,6 +616,9 @@ struct Agent {
     thread_id: uuid::Uuid,
     query_id: uuid::Uuid,
 
+    query: String,
+    branch: Option<String>,
```

Comment:
Can we move `branch` to `Conversation`? If a single conversation can span multiple branches, switching between exchanges, then can we move it to `Exchange` instead?

Reply:
@ggordonhall ?

Reply:
A `Conversation` can span and switch between multiple branches, so we should move it to `Exchange`

Reply:
@calyptobai Have pushed this, can you check it out?

Reply:
I've slightly simplified this in [3d772c3](https://github.com/BloopAI/bloop/pull/639/commits/3d772c307205938ca18ea42c5ad0f34ed4ce7490), we now rely on the parsed `SemanticQuery` stored on `Exchange`.